daml-foundations/daml-tools/da-hs-daml-cli/DA/Cli/Damlc/Test.hs

```diff
@@ -35,62 +35,65 @@ import System.FilePath
 import qualified Text.XML.Light as XML
 
 
-newtype ColorTestResults = ColorTestResults{getColorTestResults :: Bool}
+newtype UseColor = UseColor {getUseColor :: Bool}
 
 -- | Test a DAML file.
-execTest :: [FilePath] -> ColorTestResults -> Maybe FilePath -> Compiler.Options -> IO ()
-execTest inFiles colorTestResults mbJUnitOutput cliOptions = do
+execTest :: [FilePath] -> UseColor -> Maybe FilePath -> Compiler.Options -> IO ()
+execTest inFiles color mbJUnitOutput cliOptions = do
     loggerH <- getLogger cliOptions "test"
     opts <- Compiler.mkOptions cliOptions
-    -- TODO (MK): For now the scenario service is only started if we have an event logger
-    -- so we insert a dummy event logger.
-    let eventLogger _ = pure ()
-    Managed.with (Compiler.newIdeState opts (Just eventLogger) loggerH) $ \hDamlGhc -> do
-        liftIO $ Compiler.setFilesOfInterest hDamlGhc inFiles
-        mbDeps <- liftIO $ CompilerService.runAction hDamlGhc $ fmap sequence $ mapM CompilerService.getDependencies inFiles
-        depFiles <- maybe (reportDiagnostics hDamlGhc "Failed get dependencies") pure mbDeps
-        let files = Set.toList $ Set.fromList inFiles `Set.union`  Set.fromList (concat depFiles)
+    let eventLogger (EventFileDiagnostics diag) = printDiagnostics $ fdDiagnostics diag
+        eventLogger _ = return ()
+    Managed.with (Compiler.newIdeState opts (Just eventLogger) loggerH) $ \h -> do
         let lfVersion = Compiler.optDamlLfVersion cliOptions
-        case mbJUnitOutput of
-            Nothing -> testStdio lfVersion hDamlGhc files colorTestResults
-            Just junitOutput -> testJUnit lfVersion hDamlGhc files junitOutput
+        _ <- testRun h inFiles lfVersion color mbJUnitOutput
```

Comment:
```suggestion
        testRun h inFiles lfVersion color mbJUnitOutput
```

Reply:
Thanks! This worked this way at one point in time, and I forgot to update